### ISSUE TITLE
Segmentation fault in LoadSample with QtMultimedia

### ISSUE THREAD

**Issue by akien-mga** [Andrettin/Wyrmgus#158]
- Version: 4.0.5
- OS: Mageia 8

When starting `wyrmgus -d .` from a Wyrmsun git clone (to work around #157), I get this crash:
```
(gdb) r -d .
Starting program: /usr/games/wyrmgus -d .
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7ffff0a85640 (LWP 484112)]
[New Thread 0x7fffeb154640 (LWP 484116)]
[New Thread 0x7fffea953640 (LWP 484117)]
[New Thread 0x7fffea152640 (LWP 484118)]
[New Thread 0x7fffd649f640 (LWP 484119)]
[New Thread 0x7fffd5c9e640 (LWP 484120)]
[New Thread 0x7fffd549d640 (LWP 484121)]
[New Thread 0x7fffd4c9c640 (LWP 484122)]
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process
GStreamer; Unable to start decoding process

Thread 4 "wyrmgus" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffea953640 (LWP 484117)]
0x00007fffe8a50970 in gst_poll_get_read_gpollfd () from /lib64/libgstreamer-1.0.so.0
(gdb) bt
#0  0x00007fffe8a50970 in gst_poll_get_read_gpollfd () at /lib64/libgstreamer-1.0.so.0
#1  0x00007fffe8a1498a in gst_bus_constructed () at /lib64/libgstreamer-1.0.so.0
#2  0x00007fffe898a1d7 in g_object_new_internal () at /lib64/libgobject-2.0.so.0
#3  0x00007fffe898b48d in g_object_new_with_properties () at /lib64/libgobject-2.0.so.0
#4  0x00007fffe898bf41 in g_object_new () at /lib64/libgobject-2.0.so.0
#5  0x00007fffe8a14f53 in gst_bus_new () at /lib64/libgstreamer-1.0.so.0
#6  0x00007fffe8a47f6d in gst_pipeline_init () at /lib64/libgstreamer-1.0.so.0
#7  0x00007fffe89a3437 in g_type_create_instance () at /lib64/libgobject-2.0.so.0
#8  0x00007fffe898a0f5 in g_object_new_internal () at /lib64/libgobject-2.0.so.0
#9  0x00007fffe898b48d in g_object_new_with_properties () at /lib64/libgobject-2.0.so.0
#10 0x00007fffe898bf41 in g_object_new () at /lib64/libgobject-2.0.so.0
#11 0x00007fffe8a296aa in gst_element_factory_create () at /lib64/libgstreamer-1.0.so.0
#12 0x00007fffe8a298c0 in gst_element_factory_make () at /lib64/libgstreamer-1.0.so.0
#13 0x00007fffe8cb465d in QGstreamerAudioDecoderSession::QGstreamerAudioDecoderSession(QObject*) () at /usr/lib64/qt5/plugins/mediaservice/libgstaudiodecoder.so
#14 0x00007fffe8cb3b54 in QGstreamerAudioDecoderService::QGstreamerAudioDecoderService(QObject*) () at /usr/lib64/qt5/plugins/mediaservice/libgstaudiodecoder.so
#15 0x00007fffe8cb51f1 in QGstreamerAudioDecoderServicePlugin::create(QString const&) () at /usr/lib64/qt5/plugins/mediaservice/libgstaudiodecoder.so
#16 0x00007ffff76f2a4f in QPluginServiceProvider::requestService(QByteArray const&, QMediaServiceProviderHint const&) () at /lib64/libQt5Multimedia.so.5
#17 0x00007ffff77087a2 in QAudioDecoder::QAudioDecoder(QObject*) () at /lib64/libQt5Multimedia.so.5
#18 0x0000000000823a17 in wyrmgus::sample::decode(std::filesystem::__cxx11::path const&) ()
#19 0x000000000082441c in wyrmgus::sample::sample(std::filesystem::__cxx11::path const&) ()
#20 0x0000000000824503 in LoadSample(std::filesystem::__cxx11::path const&) ()
#21 0x000000000081d4e8 in wyrmgus::sound::initialize() ()
#22 0x000000000081d838 in RegisterSound(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::filesystem::__cxx11::path, std::allocator<std::filesystem::__cxx11--Type <RET> for more, q to quit, c to continue without paging--
::path> > const&) ()
#23 0x0000000000818b8d in CclMakeSound(lua_State*) ()
#24 0x00007ffff7f76be8 in luaD_precall () at /lib64/liblua.so.5.1
#25 0x00007ffff7f81575 in luaV_execute () at /lib64/liblua.so.5.1
#26 0x00007ffff7f77175 in luaD_call () at /lib64/liblua.so.5.1
#27 0x00007ffff7f764c8 in luaD_rawrunprotected () at /lib64/liblua.so.5.1
#28 0x00007ffff7f7731d in luaD_pcall () at /lib64/liblua.so.5.1
#29 0x00007ffff7f73138 in lua_pcall () at /lib64/liblua.so.5.1
#30 0x00000000008df5cb in LuaCall(int, int, bool) ()
#31 0x00000000008df6a0 in LuaLoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#32 0x00000000008df8fb in CclLoad(lua_State*) ()
#33 0x00007ffff7f76be8 in luaD_precall () at /lib64/liblua.so.5.1
#34 0x00007ffff7f81575 in luaV_execute () at /lib64/liblua.so.5.1
#35 0x00007ffff7f77175 in luaD_call () at /lib64/liblua.so.5.1
#36 0x00007ffff7f764c8 in luaD_rawrunprotected () at /lib64/liblua.so.5.1
#37 0x00007ffff7f7731d in luaD_pcall () at /lib64/liblua.so.5.1
#38 0x00007ffff7f73138 in lua_pcall () at /lib64/liblua.so.5.1
#39 0x00000000008df5cb in LuaCall(int, int, bool) ()
#40 0x00000000008df6a0 in LuaLoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#41 0x00000000008df8fb in CclLoad(lua_State*) ()
#42 0x00007ffff7f76be8 in luaD_precall () at /lib64/liblua.so.5.1
--Type <RET> for more, q to quit, c to continue without paging--
#43 0x00007ffff7f81575 in luaV_execute () at /lib64/liblua.so.5.1
#44 0x00007ffff7f77175 in luaD_call () at /lib64/liblua.so.5.1
#45 0x00007ffff7f764c8 in luaD_rawrunprotected () at /lib64/liblua.so.5.1
#46 0x00007ffff7f7731d in luaD_pcall () at /lib64/liblua.so.5.1
#47 0x00007ffff7f73138 in lua_pcall () at /lib64/liblua.so.5.1
#48 0x00000000008df5cb in LuaCall(int, int, bool) ()
#49 0x00000000008df6a0 in LuaLoadFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#50 0x00000000008df7e1 in LoadCcl(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
    ()
#51 0x000000000092b492 in stratagusMain(int, char**) ()
#52 0x00000000008b237d in std::thread::_State_impl<std::thread::_Invoker<std::tuple<main::{lambda()#1}> > >::_M_run() ()
#53 0x00007ffff5fbfed0 in execute_native_thread_routine () at /lib64/libstdc++.so.6
#54 0x00007ffff7e10dea in start_thread () at /lib64/libpthread.so.0
#55 0x00007ffff5ccfa3f in clone () at /lib64/libc.so.6
```

I have:
```
lib64qt5multimedia5-5.15.2-1.mga8
lib64gstreamer1.0_0-1.18.2-1.mga8
```

**Comment by Andrettin** [comment]
Thanks for the report! I've replaced the Qt sound decoding in the current master with SDL_mixer's, because Qt's was entirely dependent on what the OS provided. So this issue should be fixed in the next version.

**Comment by akien-mga** [comment]
Seems fixed in 4.1.2, thanks!

BTW, is there a specific reason why you're using the old SDL / SDL_mixer 1.2 instead of 2.0.x? Is it because of guichan?

**Comment by Andrettin** [comment]
That's great to hear!

The only reason is because the migration process to SDL 2.0 is not entirely painless, as a substantial part of the API changed. I intend to do so at some point though, as SDL 2.0 is a definite improvement.

Guichan should not be a problem for that, as in Stratagus itself they moved to SDL 2.0 some time ago, and they still use guichan.
